### PR TITLE
chore: pin to flutter 3.3.10

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hbuchel — APPROVED** [review]
Verified successful build in my own branch with these changes: https://updateflutterversion.dvmvffzts1tcu.amplifyapp.com/flutter/connected-components/authenticator

**Review by wlee221 — APPROVED** [review]

**Review by 0618 — APPROVED** [review]

File: amplify.yml [hunk 1]
```diff
@@ -10,7 +10,8 @@ applications:
             - nvm use 16
             - node -v
             - export FLUTTER_HOME=${HOME}/sdks/flutter
-            - git clone -b stable --depth 1 https://github.com/flutter/flutter.git ${FLUTTER_HOME}
+            # pin to flutter v3.3.10
```
Comment (calebpollman) [inline amplify.yml:13]: Is this a long term fix?

Comment (Jordan-Nelson) [reply amplify.yml:13]: Yes. I messaged you offline. If you have any other questions, lets jump on a call.

**Review by calebpollman — APPROVED** [review]
🚢

**Pull request merged**